x/bridge/keeper/assets_locked.go

```diff
@@ -97,6 +97,20 @@ func (k Keeper) AcceptAssetsLocked(
 		}
 
 		if bytes.Equal(event.TokenBytes(), sourceBTCToken) {
+			// in cases of BTC mint, we need to ensure that
+			// the funds are not sent to a blocked address first,
+			// in case they are blocked, then not minting should happen,
+			// the funds will be locked in the bridge.
+			if _, ok := k.blockedAddrs[recipient.String()]; ok {
+				ctx.Logger().Warn(
+					"BTC deposit recipient is a blocked address; "+
+						"AssetsLocked event skipped",
+					"eventSequence", event.Sequence,
+				)
+
+				continue
+			}
```

Comment:
Thoughts on placing it one level higher before the `if`, just after we parse `recipient`? This way, we also protect ERC20 bridging. This is not a problem now but there is a potential vuln if the ERC20 token would do a BTC transfer as part of the `mint` call. Moreover, this behavior will be more consistent as we won't bridge anything targeting blocked addresses, without distinguishing the asset.

Let's remember about adjusting tests and comments if we do this change.

Reply:
Yep, can do. I looked into it as mentioned in the ticket and I couldn't see how it would happen to be fair, still not sure how it could be an issue with the ERC20, but I'll believe you that one.

Indeed, I'll add tests to cover ERC20 as well.

Reply:
Yeah I'm not saying it's problematic right now. There is a small chance it **may** be problematic in the future (as mentioned in the example scenario above) so better to cover it just in case given this isn't a big lift.

Reply:
yep. sounds good.

Reply:
addressed in 7fa6a9b012cd4aca48bc9112ccca3e9a6871ee71.